src/main/java/prefs/PanelInfo.java

```diff
@@ -0,0 +1,34 @@
+package prefs;
+
+public class PanelInfo {
+
+    private String name;
+    private String filter;
```

Comment:
This class can be immutable (final fields, no getter/setter boilerplate). It won't be changed unless you create a new panel.